app/models/landable/traffic/attribution.rb

```diff
@@ -28,6 +28,8 @@ def transform(parameters)
 
         def lookup(parameters)
           where(transform(parameters)).first_or_create
+        rescue ActiveRecord::RecordNotUnique
+          retry
```

Comment:
It won't loop infinitely. The only time that error would ever be thrown is because the record was already inserted by a separate process. The retry will then attempt again to select the record. 

More specifically, #first_or_create is a non-atomic method that first attempts to select the record, then inserts if the record isn't found. A single select retry after "record not unique" being thrown will always return the record in question.